File: springboot/open-api/src/main/java/com/openapi/domain/dto/ws/response/InstructionMcpItem.java
package com.openapi.domain.dto.ws.response;

import com.openapi.domain.constant.ws.Instruction;
import com.openapi.domain.dto.ws.base.CommonResultDto;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.Map;

/**
 * {@link InstructionListResponse#getInstructions()} 中 {@link Instruction#MCP} 列表项，
 * 字段语义对齐 {@link ControlCommandResponse}（不含下行 {@link ControlCommandResponse#getTarget()}），
 * 并增加编排用 {@link #index}、执行路由等。
 */
@EqualsAndHashCode(callSuper = true)
@Data
public class InstructionMcpItem extends CommonResultDto {

    private Instruction type = Instruction.MCP;

    private Integer index;

    /**
     * 执行侧路由，如 android_local / rk_device（与 {@link ControlCommandResponse#getTarget()} 的推送目标语义不同）。
     */
    private String target;

    private String deviceId;

    private String commandId;

    private String command;

    private Map<String, String> params;
}

File: project/app/android/app/src/main/java/com/magicvector/domain/dto/ws/response/InstructionMcpItem.kt
package com.magicvector.domain.dto.ws.response

import com.magicvector.domain.constant.ws.Instruction
import com.magicvector.domain.dto.ws.base.CommonResultDto

/**
 * [InstructionListResponse.instructions] 中 [Instruction.MCP] 列表项；字段语义对齐 [ControlCommandResponse]（不含其 WS 下行 [ControlCommandResponse.target]）。
 */
data class InstructionMcpItem(
    val index: Int,
    val type: Instruction = Instruction.MCP,
    /** 执行侧路由，如 android_local / rk_device */
    val target: String,
    val deviceId: String? = null,
    val commandId: String,
    val command: String,
    val params: Map<String, String>,
    requestId: String = "",
    code: Int = 0,
    message: String = "",
) : CommonResultDto(requestId, code, message)
